batch/batch/front_end/query/query_v2.py

```diff
@@ -125,40 +125,42 @@ def parse_list_batches_query_v2(user: str, q: str, last_batch_id: Optional[int])
         where_conditions.append(f'({cond})')
         where_args += args
 
-    sql = f"""
-SELECT batches.*,
-  cancelled_t.cancelled IS NOT NULL AS cancelled,
-  job_groups_n_jobs_in_complete_states.n_completed,
-  job_groups_n_jobs_in_complete_states.n_succeeded,
-  job_groups_n_jobs_in_complete_states.n_failed,
-  job_groups_n_jobs_in_complete_states.n_cancelled,
-  cost_t.cost, cost_t.cost_breakdown
-FROM job_groups
-LEFT JOIN batches ON batches.id = job_groups.batch_id
-LEFT JOIN billing_projects ON batches.billing_project = billing_projects.name
-LEFT JOIN job_groups_n_jobs_in_complete_states ON job_groups.batch_id = job_groups_n_jobs_in_complete_states.id AND job_groups.job_group_id = job_groups_n_jobs_in_complete_states.job_group_id
-LEFT JOIN LATERAL (
-  SELECT 1 AS cancelled
-  FROM job_group_self_and_ancestors
-  INNER JOIN job_groups_cancelled
-    ON job_group_self_and_ancestors.batch_id = job_groups_cancelled.id AND
-      job_group_self_and_ancestors.ancestor_id = job_groups_cancelled.job_group_id
-  WHERE job_groups.batch_id = job_group_self_and_ancestors.batch_id AND
-    job_groups.job_group_id = job_group_self_and_ancestors.job_group_id
-) AS cancelled_t ON TRUE
-STRAIGHT_JOIN billing_project_users ON batches.billing_project = billing_project_users.billing_project
-LEFT JOIN LATERAL (
-  SELECT COALESCE(SUM(`usage` * rate), 0) AS cost, JSON_OBJECTAGG(resources.resource, COALESCE(`usage` * rate, 0)) AS cost_breakdown
-  FROM (
+    sql = f"""\
+SELECT batches.*
+     , cancelled_t.cancelled IS NOT NULL AS cancelled
+     , job_groups_n_jobs_in_complete_states.n_completed
+     , job_groups_n_jobs_in_complete_states.n_succeeded
+     , job_groups_n_jobs_in_complete_states.n_failed
+     , job_groups_n_jobs_in_complete_states.n_cancelled
+     , cost_t.cost
+     , cost_t.cost_breakdown
+FROM batches IGNORE INDEX (batches_deleted)
+STRAIGHT_JOIN billing_projects
+  ON batches.billing_project = billing_projects.name
+STRAIGHT_JOIN billing_project_users
+  ON batches.billing_project = billing_project_users.billing_project
+STRAIGHT_JOIN job_groups
+  ON job_groups.batch_id = batches.id
+STRAIGHT_JOIN job_groups_n_jobs_in_complete_states
+  ON job_groups.batch_id     = job_groups_n_jobs_in_complete_states.id
+ AND job_groups.job_group_id = job_groups_n_jobs_in_complete_states.job_group_id
+LEFT JOIN (SELECT *, 1 AS cancelled FROM job_groups_cancelled) AS cancelled_t
+  ON job_groups.batch_id     = cancelled_t.id
+ AND job_groups.job_group_id = cancelled_t.job_group_id
+STRAIGHT_JOIN LATERAL (
+  WITH resource_costs AS (
     SELECT resource_id, CAST(COALESCE(SUM(`usage`), 0) AS SIGNED) AS `usage`
     FROM aggregated_job_group_resources_v3
-    WHERE job_groups.batch_id = aggregated_job_group_resources_v3.batch_id AND job_groups.job_group_id = aggregated_job_group_resources_v3.job_group_id
+    WHERE batch_id = batches.id
```

Comment:
Do we not need to match on the job group id here anymore?

Reply:
I think you're right an I misunderstood how data was aggregated in this table.

Reply:
Yep, thanks for calling that out, fixed!